runtime/altair/src/migrations.rs

```diff
@@ -166,3 +169,145 @@ mod asset_registry {
+
+			let old_state = OldState::decode(&mut state.as_ref())
+				.map_err(|_| "Error decoding pre-upgrade state")?;
+
+			let new_total_issuance =
+				orml_tokens::TotalIssuance::<Runtime>::get(NEW_AUSD_CURRENCY_ID);
+
+			ensure!(
+				old_state.total_issuance == new_total_issuance,
+				"The old AUSD issuance differs from the new one"
+			);
+
+			for (account, account_data) in old_state.entries {
+				ensure!(
+					OrmlTokens::accounts(&account, NEW_AUSD_CURRENCY_ID) == account_data.clone(),
+					"The account data under the new AUSD Currency does NOT match the old one"
+				);
+			}
+
+			return Ok(());
+		}
+
+		fn on_runtime_upgrade() -> Weight {
+			use frame_support::traits::tokens::fungibles::Mutate;
+
+			let mut migrated_entries = 0;
+
+			// Burn all AUSD tokens under the old CurrencyId and mint them under the new one
+			orml_tokens::Accounts::<Runtime>::iter()
+				.filter(|(_, old_currency_id, _)| *old_currency_id == DEPRECATED_AUSD_CURRENCY_ID)
+				.for_each(|(account, _, account_data)| {
+					let balance = account_data.free;
+					// Burn the amount under the old, hardcoded CurrencyId
+					<orml_tokens::Pallet<Runtime> as Mutate<AccountId>>::burn_from(
+						DEPRECATED_AUSD_CURRENCY_ID,
+						&account,
+						balance,
+					)
+					.map_err(|e| {
+						log::error!(
+							"Failed to call burn_from({:?}, {:?}, {balance}): {:?}",
+							DEPRECATED_AUSD_CURRENCY_ID,
+							account,
+							e
+						)
+					})
+					.ok();
+					// // Now mint the amount under the new CurrencyID
+					<orml_tokens::Pallet<Runtime> as Mutate<AccountId>>::mint_into(
+						NEW_AUSD_CURRENCY_ID,
+						&account,
+						balance,
+					)
+					.map_err(|e| {
+						log::error!(
+							"Failed to mint_into burn_from({:?}, {:?}, {balance}): {:?}",
+							NEW_AUSD_CURRENCY_ID,
+							account,
+							e
+						)
+					})
+					.ok();
+
+					migrated_entries += 1;
+				});
+
+			// Approximate weight given for every entry migration there are two calls being
+			// made, so counting one read and one write per call. Given that we have very
+			// few entries to be migrated, the specific value isn't quite relevant and we
+			// could just set this to a high-enough value.
+			<Runtime as frame_system::Config>::DbWeight::get()
+				.reads_writes(migrated_entries * 2, migrated_entries * 2)
+		}
+	}
+}
+
+mod pool_system {
+	use pallet_pool_system::pool_types::PoolDetails;
+
+	use super::*;
+
+	pub struct MigrateAUSDPools;
+
+	impl OnRuntimeUpgrade for MigrateAUSDPools {
+		fn on_runtime_upgrade() -> Weight {
+			pallet_pool_system::Pool::<Runtime>::translate(
+				|_, mut details: PoolDetails<CurrencyId, _, _, _, _, _, _, _, _>| {
+					if details.currency == DEPRECATED_AUSD_CURRENCY_ID {
+						details.currency = NEW_AUSD_CURRENCY_ID;
+					}
+
+					Some(details)
+				},
+			);
+
+			let n = pallet_pool_system::Pool::<Runtime>::iter().count() as u64;
+			<Runtime as frame_system::Config>::DbWeight::get().reads_writes(n, n)
+		}
+	}
+}
```

Comment:
ℹ️  This is a super simple, idempotent migration. It will only run on Altair and apply to the single pool we have there.

Reply:
Would it be possible to add the `try-runtime` pre and post upgrade hooks? I understand it seems unnecessary in this scenario but I think we should NEVER run a migration without pre and post hooks. Otherwise there is no way to test it apart from locally running the migration. Since executing `try-runtime` is part of the release process, it should be enforced.

Reply:
Done ✅